src/util/amount.rs

```diff
@@ -0,0 +1,362 @@
+#![allow(missing_docs)]
+
+use std::error;
+use std::fmt;
+use std::ops;
+use std::str::FromStr;
+
+pub trait Precision: Default + Clone {
+    /// The power of 10 a satoshi should be multiplied with to gain this precision.
+    fn scale() -> i32;
+    fn denom() -> &'static str;
+}
+
+#[derive(Default, Clone)]
+pub struct Bitcoin();
+impl Precision for Bitcoin {
+    fn scale() -> i32 {
+        -8
```

Comment:
> /// The power of 10 a satoshi should be multiplied with to gain this precision. 

Are you sure it shouldn't be `8` since `sat * 10^8 = btc`?

Reply:
Oh yeah I changed that and didn't change doc. But I think actually having `8` and `-3` makes more sense.